app/src/main/java/de/westnordost/streetcomplete/data/quest/QuestController.kt

```diff
@@ -142,7 +142,7 @@ import kotlin.collections.ArrayList
 
         // first remove old tags
         for ((key, value) in previousTags) {
-            val isOkToRemove = KEYS_THAT_SHOULD_NOT_BE_REMOVED_WHEN_SHOP_IS_REPLACED.none { it.matches(key) }
+            val isOkToRemove = KEYS_THAT_SHOULD_BE_REMOVED_WHEN_SHOP_IS_REPLACED.none { !it.matches(key) }
```

Comment:
For some reason GitHub's "suggest changes" feature doesn't appear to be available right now, but I would suggest this instead:

```
val isOkToRemove = KEYS_THAT_SHOULD_BE_REMOVED_WHEN_SHOP_IS_REPLACED.any { it.matches(key) }
```

Reply:
thanks! changed in 1df4132118baf0f263956f99681e6189e06613d6